libs/portfolio/domain/src/lib/command-handlers/unarchive-portfolio.command-handler.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { EventStore } from '@pl-oss/domain-core';
+import { PortfolioAggregate } from '../aggregate/portfolio.aggregate';
+import { PortfolioAggregateRepository } from '../aggregate-repository/portfolio.aggregate-repository';
+import { UnarchivePortfolioCommand } from '../command/unarchive-portfolio.command';
+import { UnarchivePortfolioCommandHandler } from './unarchive-portfolio.command-handler';
+
+jest.mock('../aggregate-repository/portfolio.aggregate-repository');
+
+describe('UnarchivePortfolioCommandHandler', () => {
+  const id = 'id';
+
+  let testPortfolioAggregate: PortfolioAggregate;
+  let testPortfolioAggregateRepository: PortfolioAggregateRepository;
+  let testUnarchivePortfolioCommand: UnarchivePortfolioCommand;
+  let testUnarchivePortfolioCommandHandler: UnarchivePortfolioCommandHandler;
+
+  beforeEach(() => {
+    testPortfolioAggregate = new PortfolioAggregate();
+    testPortfolioAggregateRepository = new PortfolioAggregateRepository({} as EventStore);
+    testUnarchivePortfolioCommand = new UnarchivePortfolioCommand(id);
+    testUnarchivePortfolioCommandHandler = new UnarchivePortfolioCommandHandler(
+      testPortfolioAggregateRepository,
+    );
+
+    testPortfolioAggregate.isArchived = true;
+    jest.spyOn(testPortfolioAggregateRepository, 'getById').mockResolvedValue(testPortfolioAggregate);
+    jest.spyOn(testPortfolioAggregateRepository, 'save').mockResolvedValue(undefined);
+    jest.spyOn(testPortfolioAggregate, 'unarchive');
```

Comment:
Opening suggestion for a discussion.
Instead of setting it up with `archived` initial value what if we resolve the `unarchive` spy with something and test if unarchive and save methods are being called?

```suggestion
    jest.spyOn(testPortfolioAggregateRepository, 'getById').mockResolvedValue(testPortfolioAggregate);
    jest.spyOn(testPortfolioAggregateRepository, 'save').mockResolvedValue(undefined);
    jest.spyOn(testPortfolioAggregate, 'unarchive').mockReturnValue(undefined);
```

My rationale is that in the handler doesn't know the precondition or postcondition of aggregate. All it knows is to get it form repo, call unarchive and save it. lmk your thoughts

Reply:
We should not test the behaviour of archive or unarchive methods. Please test the call and we are done. Otherwise, we are coupling the business logic here.

Moreover, the business is already tested in PortfolioAggregate's test.

Reply:
So I agree with Shail's comment.